### ISSUE TITLE
Closure validation

### ISSUE THREAD

**Issue by KKSzymanowski** [laravel/internals#492]
For simple, single-use, custom validation rules it seems to me like an overkill to extend the Validator as described in the docs. I'd like to propose following functionality:
```php
class FooRequest extends FormRequest
{
    // ...

    public function rules()
    {
        'password' => [
            'required',
            'customRule' => function ($attribute, $value, $parameters, $validator) {
                return some_validation_logic($value);
            }
        ];
    }

    public function messages()
    {
        return [
            'password.required' => 'Foo',
            'password.customRule' => 'Bar',
        ];
    }
}
```
What do you think?

**Comment by KKSzymanowski** [comment]
@taylorotwell Are you interested in this? 
I think `$attribute` and `$parameters` are redundant, because closure is created for one attribute and there's no way to pass the parameters so it would always be an empty array.

**Comment by Lidbetter** [comment]
I wanted something like this - I have implemented it as such:

```
class ClosureRule implements Rule
{
    protected $fn;
    protected $message = 'The :attribute input is invalid';

    public function __construct(Closure $closure)
    {
        $this->fn = $closure;
    }

    public function passes($attribute, $value)
    {
        $result = ($this->fn)($attribute, $value);

        if(is_string($result)) {
            $this->message = $result;
        }

        return ($result === true);
    }

    public function message()
    {
        return $this->message;
    }
}
```

```
// useage
$this->validate($request, [
    'password' => [
            'required',
            'customRule' => Rule::closure(function ($attribute, $value) {
                return some_validation_logic($value);
            }),
        ],
]);
```

**Comment by KKSzymanowski** [comment]
In has been implemented in the core since 5.5:
https://github.com/laravel/framework/blob/5.5/src/Illuminate/Validation/ClosureValidationRule.php
